### PR TITLE
👨‍💻 Updates to `Makefile` 

### PR DESCRIPTION
**Pull request by Gary-H9**
This PR does a number of things: 

- ⬇️ Downgrade `mypy` to resolve issues
- ✏️ Capitalise header
- ➕ Add `django_extensions`
- 👶 Create `.env.local`
- 👨‍💻 Add `db-*` commands to Makefile
- 📌 Pin `govuk-frontend`

Most importantly this edits the `Makefile` to add `db-*` commands. The inclusion of the following lines pulls variables from the `.env.local` command.  
```bash
-include .env.local
export
...
```
The addition of `django_extensions`  was needed to create the `make db-drop` command.

Additionally I've added the following which are self-explanatory:

```bash
db-migrate:
	python manage.py migrate

db-connect:
	psql --username=controlpanel --host=localhost --dbname=controlpanel

db-drop:
	python manage.py reset_db

serve:
	python manage.py runserver
```

The contents of the `.env.local` file are for development purposes only.

### CODE REVIEW

File: controlpanel/settings/common.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "django.contrib.sessions",
     "django.contrib.messages",
     "django.contrib.staticfiles",
+    "django_extensions",
```
Comment (ymao2) [inline controlpanel/settings/common.py:49]: hmm,  have we installed this package?  🤔

File: requirements.dev.txt [hunk 2]
```diff
@@ -3,5 +3,5 @@ black==23.11.0
 django-stubs[compatible-mypy]==4.2.6
 flake8==6.1.0
 isort==5.12.0
-mypy==1.7.0
+mypy==1.6.0
```
Comment (ymao2) [inline requirements.dev.txt:6]: is this intended to lower the version?

File: package.json [hunk 3]
```diff
@@ -14,6 +14,6 @@
     "sass": "^1.69.5"
   },
   "dependencies": {
-    "govuk-frontend": "^4.7.0"
+    "govuk-frontend": "4.7.0"
```
Comment (ymao2) [inline package.json:17]: could you revert this change too

Comment (jacobwoffenden) [reply package.json:17]: I'd argue we should pinning exact versions and letting Dependabot track package updates as it does already

https://github.com/ministryofjustice/data-platform-control-panel/blob/13d86aaa658d3e560833ffc6df1d2b0c7e4ddc4a/.github/dependabot.yml#L20-L23

File: controlpanel/settings/common.py (hunk 1 above)
Comment (Gary-H9) [reply controlpanel/settings/common.py:49]: To be looked at at a later date

**Review by ymao2 — APPROVED** [review]
LGTM :)
